tests/ConsensusFileGenerator_test.py

```diff
@@ -46,3 +47,11 @@ def test__get_history_ids_for_variant(self, _, variant_info, expected):
         observed = ConsensusFileGenerator._get_history_ids_for_variant(variant_id, chromosome, pos, ref, alt, gene,
                                                                        variant_type)
         self.assertEqual(observed, expected)
+
+    def test__check_alternative_history(self):
```

Comment:
If I understand correctly you have the following cases:
1. before: variant is correct and has transcript:cDNA after: variant is correct
2. before: variant is correct and has no transcript:cDNA after: variant is correct
3. before: variant is incorrect and has transcript:cDNA after: new correct variant
4. before: variant is incorrect and has transcript:cDNA after: merged with other existing correct variant
5. before: variant is incorrect and has no transcript:cDNA after: new correct variant

Are are of them covered by the unit tests?

Reply:
You are correct (except that in case 3 you probably meant "has no transcript:cDNA"). This unit test is only about the alternate history but I agree that I could test for these cases elsewhere. I'll look into that.